### DISCUSSION TITLE
Using tRPC on several windows.

### DISCUSSION THREAD

**Discussion by justas-dovydaitis** [awohletz/electron-prisma-trpc-example#3] [Q&A] [answered]
Hi, I've just got stuck in a situation where I need to share a tRPC connection between two windows.

I am trying to update some data and invalidate the query on the main window and expecting another window to automatically fetch invalidated data. Now I can load data using tRPC, but it only updates on a window that made changes.
Is this kind of behavior possible to achieve at all?

@awohletz if you could help me I would be more than happy.

**Comment by awohletz** [comment, marked as answer]
I see a couple ways to do that:
1. Pass the refetchInterval and refetchIntervalInBackground options to the trpc useQuery hook. It will poll at an interval to fetch latest data.
2. Or send a signal from Main to all Renderer processes when the data in question changes. In the Renderer code, invalidate the query or use the `refetch` function of useQuery to refetch the data.

**Reply by justas-dovydaitis** [reply]
Thanks for your response, appreciate it.

I am currently using `refetchInterval` but it does not seem legit since I need to re-fetch data at least once a second to keep it updated in UI. I was just expecting there's some trick with tRPC that I do not know yet... 😆 

Anyway, good advice - sending signals to renderer processes about data changes will make it way more efficient. Thanks!
